### PR TITLE
Added visibility toggle for users.

### PR DESCRIPTION
**Pull request by Elland**
screen shot 2017-06-26 at 15 04 16

<br>

screen shot 2017-06-26 at 15 05 01

### CODE REVIEW

File: Toshi/Controllers/Settings/ProfileVisibilityController.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        self.visibilitySwitch.isOn = TokenUser.current?.isPublic ?? false
+    }
+
+    @IBAction func publicStateDidChange(_ sender: UISwitch) {
+        guard let currentUser = TokenUser.current else { return }
+
+        currentUser.updatePublicState(to: sender.isOn)
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/ProfileVisibilityController.swift]: can't we just call `currentUser?.update...`

File: Toshi/Controllers/Settings/ProfileVisibilityController.swift [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import UIKit
+import SweetUIKit
+
+class ProfileVisibilityController: UITableViewController {
+
+    @IBOutlet weak var visibilitySwitch: UISwitch!
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/ProfileVisibilityController.swift]: fileprivate

File: Toshi/Controllers/Settings/SettingsController.swift [hunk 3]
```diff
@@ -246,6 +246,11 @@ open class SettingsController: UITableViewController {
             case 0:
                 break // change currency
             case 1:
+                let storyboard = UIStoryboard(name: "ProfileVisibilityController", bundle: nil)
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/SettingsController.swift]: StoryBoard named **ProfileVisibilityController** ?

File: Toshi/Controllers/Settings/SettingsController.swift [hunk 4]
```diff
@@ -246,6 +246,11 @@ open class SettingsController: UITableViewController {
...
+                guard let profileVisiblityController = storyboard.instantiateInitialViewController() else { fatalError() }
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/SettingsController.swift]: i think we should handle this, not crash;
it is better to do nothing than crash

**Comment by yuliaveres** [comment]
As a general comment:
Where is description, definitely need this. (maybe look on Android as an example)
As tableView section footer text

File: Toshi/Controllers/Settings/ProfileVisibilityController.swift (hunk 2 above)
Comment (Elland) [inline Toshi/Controllers/Settings/ProfileVisibilityController.swift]: It's an outlet, can it be `fileprivate`? I'm fine with not messing with generated code.

File: Toshi/Controllers/Settings/SettingsController.swift (hunk 3 above)
Comment (Elland) [inline Toshi/Controllers/Settings/SettingsController.swift]: Yes?

File: Toshi/Controllers/Settings/SettingsController.swift (hunk 4 above)
Comment (Elland) [inline Toshi/Controllers/Settings/SettingsController.swift]: Forgot to replace this. 🕵🏽‍♂️

File: Toshi/Controllers/Settings/ProfileVisibilityController.storyboard [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                                                        <constraint firstAttribute="height" constant="21" id="82s-aJ-Crj"/>
+                                                    </constraints>
+                                                    <fontDescription key="fontDescription" name="SFUIText-Regular" family="SF UI Text" pointSize="17"/>
+                                                    <nil key="textColor"/>
+                                                    <nil key="highlightedColor"/>
+                                                </label>
+                                                <switch opaque="NO" contentMode="scaleToFill" horizontalHuggingPriority="750" verticalHuggingPriority="750" contentHorizontalAlignment="center" contentVerticalAlignment="center" on="YES" translatesAutoresizingMaskIntoConstraints="NO" id="cJ0-XR-Wmf">
+                                                    <rect key="frame" x="310" y="6" width="51" height="31"/>
+                                                    <constraints>
+                                                        <constraint firstAttribute="width" constant="49" id="j9O-IH-8Ls"/>
```
Comment (yuliaveres) [inline Toshi/Controllers/Settings/ProfileVisibilityController.storyboard]: why u need width constraint on switcher ?

**Comment by Elland** [comment]
@yuliaveres Implemented according to latest design spec. 😞 Will check back with the designers whey they return.

File: Toshi/Controllers/Settings/SettingsController.swift (hunk 3 above)
Comment (yuliaveres) [inline Toshi/Controllers/Settings/SettingsController.swift]: maybe ProfileVisibility ? as it is storyboard, not a controller.
i think we should name the scene, not as single controller name

File: Toshi/Controllers/Settings/ProfileVisibilityController.swift (hunk 2 above)
Comment (yuliaveres) [inline Toshi/Controllers/Settings/ProfileVisibilityController.swift]: yes, it should be private, it is not messing

File: Toshi/Controllers/Settings/ProfileVisibilityController.storyboard (hunk 5 above)
Comment (Elland) [inline Toshi/Controllers/Settings/ProfileVisibilityController.storyboard]: Storyboard added this. ¯\_(ツ)_/¯

File: Toshi/Controllers/Settings/SettingsController.swift (hunk 3 above)
Comment (Elland) [inline Toshi/Controllers/Settings/SettingsController.swift]: 🤦🏽‍♂️ Of course.

File: Toshi/Controllers/Settings/ProfileVisibilityController.storyboard (hunk 5 above)
Comment (yuliaveres) [inline Toshi/Controllers/Settings/ProfileVisibilityController.storyboard]: 👀

**Comment by yuliaveres** [comment]
Can't we copy text from Android ? @Elland

**Comment by Elland** [comment]
I think so. Want to hold this out until designers come back? They might have some whole new direction for this.

**Comment by Elland** [comment]
The "hard part" is done, backend integration stuff. 👍

**Comment by yuliaveres** [comment]
i think it will be done with footer text as normally on iOS, please, if you can do it so it at least looks better and gives important info about the property.
or is that going to take much time ?

**Pull request merged**